csharp/src/Ice/IceDiscovery/Locator.cs

```diff
@@ -143,7 +144,9 @@ public ValueTask SetAdapterDirectProxyAsync(string adapterId, IObjectPrx? proxy,
                     try
                     {
                         IObjectPrx proxy = _adapters[ids.First()];
-                        proxy = proxy.Clone(IObjectPrx.Factory, adapterId: key, identity: identity);
+                        proxy = proxy.Clone(IObjectPrx.Factory,
+                                            identity: identity,
+                                            location: ImmutableArray.Create(key));
```

Comment:
Previously this Clone produce an Indirect proxy because setting the `adaterId` clear the endpoints, but that is no longer true for Ice, not clear why we did it this way, but my understanding is that now with Ice2 the proxy will still have the original endpoints.

Reply:
Fixed.